gpytorch/module.py

```diff
@@ -270,6 +278,15 @@ def sample_from_prior(self, prior_name):
             raise RuntimeError("Must provide inverse transform to be able to sample from prior.")
         setting_closure(prior.sample())
 
+    def pyro_sample_from_prior(self):
+        return _pyro_sample_from_prior(module=self, memo=None, prefix="")
+
+    def pyro_load_from_samples(self, samples_dict):
+        self._strict(False)
```

Comment:
This needs to call recursively rather than do everything from the top level, because some modules will have additional behavior (e.g., kernels will have to update batch_shape).